src/main/scala/chisel3/Driver.scala

```diff
@@ -149,7 +149,11 @@ object Driver extends BackendCompilationUtilities {
     /* create custom transforms by finding the set of transform classes associated with annotations
      * then instantiate them into actual transforms
      */
-    val transforms = circuit.annotations.map(_.transform).toSet.map { transformClass: Class[_ <: Transform] =>
+    val transforms = circuit.annotations
+                            .map(_.transform)
+                            .distinct
+                            .filterNot(_ == classOf[firrtl.Transform])
```

Comment:
So the issue here is that it's convenient for custom transforms to be included automatically if there's an Annotation that targets them. It's obvious that people do not intend for a transform to be instantiated when an annotation targets a built-in Transform that is run anyway, let alone when the targeted transform is firrtl.Transform which is the abstract superclass of all Transforms!

I'll add a comment to make this more clear to future readers, but improving the Transform auto instantiation API is beyond the scope of this PR imo.